### ISSUE TITLE
Review the accessibility of behavior in DropDown

### ISSUE THREAD

**Issue by matuzalemsteles** [liferay/clay#5041]
After making some accessibility improvements to DropDown, like #4998 which adds semantic role to correctly give context to Menu items and semantics to the DropDown trigger we also sent more improvements in #5038 which improves navigation when using virtual cursor in screen reader and adds the possibility to open DropDown using ArrowDown and Spacebar, move the focus to the first focusable DropDown element, and also correct semantics for groups but there are still some inconsistencies of behavior and shortcuts accordingly https://www.w3.org/WAI/ARIA/apg/example-index/menu-button/menu-button-links.

See the comment https://github.com/liferay/clay/pull/5038#issuecomment-1216013803 to understand more about it. This issue is mainly to understand some main behaviors that we have for years in the component that deviates from the accessibility rules.

- Tab within DropDown should not be possible
- Navigating using arrow keys should be looped

The idea here is to raise a discussion mainly to understand if this makes sense for the component mainly the tabulation that I remember was added to allow navigation using only keyboard, I think at the beginning we had put the navigation arrows in a loop and to provide for the user exiting the dropdown would be adding the tabulation I think we ended up seeing this the wrong way, the tab would probably not do navigation through the menu but close the menu and go to the next focusable element.

@marcoscv-work I would like to have your comments here too and @drakonux jfyi

**Comment by marcoscv-work** [comment]
About patterns, from issue https://github.com/liferay/clay/issues/5147:

I'm going to try to explain the difference between patterns with focus to the fist element and dont move the focus until press tab or arrows (with menu opened):

Pattern: **Focus auto move to the first item**:

1. SR reads click me, I'm a popup collapsed button:
<img width="755" alt="image" src="https://user-images.githubusercontent.com/7963804/197184534-2d380dcb-578d-4b24-aa6a-a175816461d3.png">
2. We press the button and directly the SR reads: Button (or any first item).
<img width="753" alt="image" src="https://user-images.githubusercontent.com/7963804/197184926-90c58a5b-eac3-4810-b417-62921564b8ed.png">

Pattern:  **Keep the focus on the trigger and reads its change**:

1. SR reads click me, I'm a popup collapsed button:
<img width="755" alt="image" src="https://user-images.githubusercontent.com/7963804/197184534-2d380dcb-578d-4b24-aa6a-a175816461d3.png">
2. We press the button and the SR reads the state change.
<img width="753" alt="image" src="https://user-images.githubusercontent.com/7963804/197185982-3754e4b8-b426-4705-8b0c-ad1082f71601.png">
3. Now we press arrow down or tab and we get in the first item and it is read:
<img width="753" alt="image" src="https://user-images.githubusercontent.com/7963804/197184926-90c58a5b-eac3-4810-b417-62921564b8ed.png">


### Using JAWS:

It's the same, but the state change is read shorter:
Pattern: "Key you pressed / state changed"

**Focus auto move to the first item**:
1. SR reads click me, I'm a popup collapsed button:
2. We press the button and directly the SR reads: Button (or any first item).
"Enter - Button" (the first button/link in dropdown)

1. SR reads click me, I'm a popup collapsed button:
2. We press the button and the SR reads the state change.
"Enter - Expanded"
3. Now we press arrow down or tab and we get in the first item and it is read:
"Button" (the first item in dropdown)


## Conclusion:

- For me the pattern: **Focus auto move to the first item** can be even sometimes difficult to understand.
  - The image example have a button to get back without semantic but the "jump without context" is the same if the first option is a simple link or button in a simple dropdown (not a Drilldown one).

**In case my opinion is not clear**: I definitely wouldn't forced to leave the trigger context and move forward by moving the focus to the first element, I think the Bootstrap example is the best solution.
